[2024-03-21T16:51:37.717+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spotify_to_s3_data_extraction_v1.sptoify_to_csv scheduled__2024-03-21T07:33:00+00:00 [queued]>
[2024-03-21T16:51:37.751+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spotify_to_s3_data_extraction_v1.sptoify_to_csv scheduled__2024-03-21T07:33:00+00:00 [queued]>
[2024-03-21T16:51:37.757+0000] {taskinstance.py:2193} INFO - Starting attempt 5 of 10
[2024-03-21T16:51:37.845+0000] {taskinstance.py:2217} INFO - Executing <Task(PythonOperator): sptoify_to_csv> on 2024-03-21 07:33:00+00:00
[2024-03-21T16:51:37.937+0000] {standard_task_runner.py:60} INFO - Started process 3273 to run task
[2024-03-21T16:51:37.952+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'spotify_to_s3_data_extraction_v1', 'sptoify_to_csv', 'scheduled__2024-03-21T07:33:00+00:00', '--job-id', '649', '--raw', '--subdir', 'DAGS_FOLDER/spotify_to_csvs_to_s3.py', '--cfg-path', '/tmp/tmpesqq36gi']
[2024-03-21T16:51:37.980+0000] {standard_task_runner.py:88} INFO - Job 649: Subtask sptoify_to_csv
[2024-03-21T16:51:38.298+0000] {task_command.py:423} INFO - Running <TaskInstance: spotify_to_s3_data_extraction_v1.sptoify_to_csv scheduled__2024-03-21T07:33:00+00:00 [running]> on host ca1809388db1
[2024-03-21T16:51:38.712+0000] {taskinstance.py:2513} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Nathaniel Solomon' AIRFLOW_CTX_DAG_ID='spotify_to_s3_data_extraction_v1' AIRFLOW_CTX_TASK_ID='sptoify_to_csv' AIRFLOW_CTX_EXECUTION_DATE='2024-03-21T07:33:00+00:00' AIRFLOW_CTX_TRY_NUMBER='5' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-03-21T07:33:00+00:00'
[2024-03-21T16:51:38.733+0000] {logging_mixin.py:188} INFO - The extract function has been executed and this is data in the next line
[2024-03-21T16:51:38.743+0000] {logging_mixin.py:188} INFO - Initiating the Endpoint...
[2024-03-21T16:51:39.942+0000] {logging_mixin.py:188} INFO - Extracting other available tracks
[2024-03-21T16:52:12.114+0000] {logging_mixin.py:188} INFO - Adding the values of index 0 into the appropriate Keys...
[2024-03-21T22:15:15.717+0000] {local_task_job_runner.py:211} ERROR - Heartbeat time limit exceeded!
[2024-03-21T22:15:15.894+0000] {process_utils.py:131} INFO - Sending 15 to group 3273. PIDs of all processes in the group: [3273]
[2024-03-21T22:15:15.923+0000] {process_utils.py:86} INFO - Sending the signal 15 to group 3273
[2024-03-21T22:15:16.230+0000] {taskinstance.py:2483} ERROR - Received SIGTERM. Terminating subprocesses.
[2024-03-21T22:15:16.314+0000] {taskinstance.py:2731} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 444, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 414, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 200, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 217, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/spotify_to_csvs_to_s3.py", line 110, in Spotify_to_CSV
    data['Artist_Genres'].append(sp.artist(track['artists'][0]['id'])['genres'])
  File "/home/airflow/.local/lib/python3.8/site-packages/spotipy/client.py", line 392, in artist
    return self._get("artists/" + trid)
  File "/home/airflow/.local/lib/python3.8/site-packages/spotipy/client.py", line 323, in _get
    return self._internal_call("GET", url, payload, kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/spotipy/client.py", line 266, in _internal_call
    response = self._session.request(
  File "/home/airflow/.local/lib/python3.8/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/requests/adapters.py", line 486, in send
    resp = conn.urlopen(
  File "/home/airflow/.local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 894, in urlopen
    return self.urlopen(
  File "/home/airflow/.local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 715, in urlopen
    httplib_response = self._make_request(
  File "/home/airflow/.local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 404, in _make_request
    self._validate_conn(conn)
  File "/home/airflow/.local/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1058, in _validate_conn
    conn.connect()
  File "/home/airflow/.local/lib/python3.8/site-packages/urllib3/connection.py", line 419, in connect
    self.sock = ssl_wrap_socket(
  File "/home/airflow/.local/lib/python3.8/site-packages/urllib3/util/ssl_.py", line 402, in ssl_wrap_socket
    context.load_verify_locations(ca_certs, ca_cert_dir, ca_cert_data)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2485, in signal_handler
    raise AirflowException("Task received SIGTERM signal")
airflow.exceptions.AirflowException: Task received SIGTERM signal
[2024-03-21T22:15:16.366+0000] {taskinstance.py:1149} INFO - Marking task as UP_FOR_RETRY. dag_id=spotify_to_s3_data_extraction_v1, task_id=sptoify_to_csv, execution_date=20240321T073300, start_date=20240321T165137, end_date=20240321T221516
[2024-03-21T22:15:16.431+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 649 for task sptoify_to_csv (Task received SIGTERM signal; 3273)
[2024-03-21T22:15:16.501+0000] {process_utils.py:79} INFO - Process psutil.Process(pid=3273, status='terminated', exitcode=1, started='16:51:37') (3273) terminated with exit code 1
